File: backend/app/schemas/project.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, Field


class ProjectBase(BaseModel):
    name: str = Field(
        min_length=2,
        max_length=100,
    )

    key: str = Field(
        min_length=2,
        max_length=10,
    )

    description: Optional[str] = None


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(
        default=None,
        min_length=2,
        max_length=100,
    )

    description: Optional[str] = None


class ProjectResponse(ProjectBase):
    id: int
    organization_id: int
    created_by: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )

File: frontend/src/types/project.ts
// GET projects
export interface Project {
  id: number;
  organization_id: number;
  name: string;
  key: string;
  description: string | null;
  created_by: number;
  created_at: string;
}


// create projects
export interface CreateProjectRequest {
  name: string;
  key: string;
  description?: string;
}


// update projects
export interface UpdateProjectRequest {
  name?: string;
  description?: string;
}